mlflow/tracing/processor/mlflow.py

```diff
@@ -86,10 +85,20 @@ def on_start(self, span: OTelSpan, parent_context: Optional[Context] = None):
         span.set_attribute(SpanAttributeKey.REQUEST_ID, json.dumps(request_id))
 
     def _start_trace(self, span: OTelSpan, start_time_ns: Optional[int]) -> TraceInfo:
+        from mlflow.tracking.fluent import _get_latest_active_run
+
         experiment_id = get_otel_attribute(span, SpanAttributeKey.EXPERIMENT_ID)
         metadata = {TRACE_SCHEMA_VERSION_KEY: str(TRACE_SCHEMA_VERSION)}
         # If the span is started within an active MLflow run, we should record it as a trace tag
-        if run := mlflow.active_run():
+        # Note `mlflow.active_run()` can only get thread-local active run,
+        # but tracing routine might be applied to model inference worker threads
+        # in the following cases:
+        #  - langchain model `chain.batch` which uses thread pool to spawn workers.
+        #  - MLflow langchain pyfunc model `predict` which calls `api_request_parallel_processor`.
+        # Therefore, we use `_get_global_active_run()` instead to get the active run from
+        # all threads and set it as the tracing source run.
+        latest_active_run = _get_latest_active_run()
+        if run := latest_active_run:
```

Comment:
```suggestion
        if run := _get_latest_active_run():
```


Is `latest_active_run` used after this line?